core/src/test/java/org/elasticsearch/index/query/ExistsQueryBuilderTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermRangeQuery;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.index.mapper.FieldMapper;
+import org.elasticsearch.index.mapper.internal.FieldNamesFieldMapper;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExistsQueryBuilderTest extends BaseQueryTestCase<ExistsQueryBuilder> {
+
+    @Override
+    protected ExistsQueryBuilder createEmptyQueryBuilder() {
+        return new ExistsQueryBuilder(null);
```

Comment:
I already thought about this when I introduced the PROTOTYPE constant but then found it strange, since we use the empty Builder instances created here in the tests to e.g. serialize into, so they change state. The prototypes constants IMHO should only be used to create new builders via readFrom().